### ISSUE TITLE
decorators inside singleton classes

### ISSUE THREAD

**Issue by jaredlwong** [kjleitz/adornable#9]
Why can't I add decortors inside singleton classes? I get that the error is saying `name` isn't defined for the singleton class, but why do we need it?
```ruby
require 'adornable'

class Decorators
  def self.time(context, stat:, &block)
    start_time = Time.now.utc
    begin
      return yield
    ensure
      end_time = Time.now.utc
      duration = (end_time - start_time) * 1000.0
      puts "#{stat} #{duration}ms"
    end
  end
end

class Dog
  class << self
    extend Adornable

    decorate :time, from: Decorators, stat: 'bark'
    def bark
      sleep 1.3
    end
  end
end

Dog.bark
```

```ruby
Traceback (most recent call last):
        3: from adornable.rb:16:in `<main>'
        2: from adornable.rb:17:in `<class:Dog>'
        1: from adornable.rb:20:in `singleton class'
/Users/jaredw/.rbenv/versions/2.7.6/lib/ruby/gems/2.7.0/gems/adornable-1.2.0/lib/adornable.rb:18:in `decorate': Decorator name must be provided. (Adornable::Error::InvalidDecoratorArguments)
```

```ruby
  def decorate(decorator_name, from: nil, defer_validation: false, **decorator_options)
    if Adornable::Utils.blank?(name)
      raise Adornable::Error::InvalidDecoratorArguments, "Decorator name must be provided."
    end
```

**Comment by kjleitz** [comment]
Hi @jaredlwong, sorry I took so long to respond here. I'll check this issue out later today.

**Comment by kjleitz** [comment]
@jaredlwong This issue is now fixed in version 1.2.1 :+1:
